lib/kafka/sidekiq/event_bus_submission_job.rb

```diff
@@ -2,32 +2,40 @@
 
 require 'kafka/avro_producer'
 require 'kafka/monitor'
+require 'kafka/concerns/topic'
 
 module Kafka
   class EventBusSubmissionJob
     include Sidekiq::Job
+    include Kafka::Topic
+
     sidekiq_options retry: 3, queue: 'low'
 
     # retry exhaustion
     sidekiq_retries_exhausted do |msg|
       monitor = Kafka::Monitor.new
-      topic = msg['args'].first
-      payload = msg['args'].second
+      payload = msg['args'].first
+      use_test_topic = msg['args'].second
+      topic = get_topic(use_test_topic:)
+      redacted_payload = redact_icn(payload)
 
-      monitor.track_submission_exhaustion(msg, topic, payload)
+      monitor.track_submission_exhaustion(msg, topic, redacted_payload)
     end
 
     # Performs the job of producing a message to a Kafka topic
     #
     # @param topic [String] The Kafka topic to which the message will be sent
     # @param payload [Hash] The message payload to be sent to the Kafka topic
-    def perform(topic, payload)
+    def perform(payload, use_test_topic = false) # rubocop:disable Style/OptionalBooleanParameter
```

Comment:
Should we be more specific about this i.e.`def perform(payload, use_test_topic: false)` so that its clear what true/false means?

Reply:
There's some weirdness with how sidekiq accepts parameters. I originally tried it with named parameters but couldn't get it to work. I learned that Sidekiq does not accept complex arguments parameter values. See https://github.com/sidekiq/sidekiq/issues/2372 and https://github.com/sidekiq/sidekiq/wiki/Best-Practices